src/test/store/profile-view.test.js

```diff
@@ -196,3 +201,205 @@ describe('expand all call node descendants', function() {
     ]);
   });
 });
+
+/**
+ * Naively run through all the selectors. The correctness of what they are computing
+ * should be left up to better informed unit tests. This provides some base coverage
+ * of mechanically running through the selectors in tests.
+ */
+describe('snapshots of selectors/profile-view', function() {
+  // Set up a profile that has some nice features that can show that the selectors work.
+  const {
+    profile,
+    funcNamesPerThread: [funcNames],
+  } = getProfileFromTextSamples(`
+    A A A A A A A A A
+    B B B B B B B B B
+    C C C C C C H H H
+    D D D F F F I I I
+    E E E G G G
+  `);
+  const A = funcNames.indexOf('A');
+  const B = funcNames.indexOf('B');
+  const C = funcNames.indexOf('C');
+
+  const [samplesThread] = profile.threads;
+  // Add in a thread with markers
+  const { threads: [markersThread] } = getProfileWithMarkers([
+    ['A', 0, null],
+    ['B', 1, null],
+    ['C', 2, null],
+    ['D', 3, null],
+    ['E', 4, null],
+    ['F', 5, null],
+  ]);
+  profile.threads.push(markersThread);
+  const { getState, dispatch } = storeWithProfile(profile);
+  samplesThread.name = 'Thread with samples';
+  markersThread.name = 'Thread with markers';
+  samplesThread.markers = markersThread.markers;
+  // This is a jank sample:
+  samplesThread.samples.responsiveness[4] = 100;
+  const mergeFunction = {
+    type: 'merge-function',
+    funcIndex: C,
+  };
+  dispatch(ProfileView.addTransformToStack(0, mergeFunction));
+  dispatch(ProfileView.changeExpandedCallNodes(0, [[A], [A, B]]));
+  dispatch(ProfileView.changeSelectedCallNode(0, [A, B]));
+  dispatch(ProfileView.changeSelectedMarker(0, 1));
+  dispatch(ProfileView.addRangeFilter(3, 7));
```

Comment:
My only nit is about the fact we share the same store for all cases. This isn't a big deal here because all `it` blocks don't change the store and are only reading it. But still we could have side-effects depending on how we change this testcase in the future.

Rather, I'd extract all this setup in a function `storeWithSomeOperations` (or whatever name you chose) that would return the store, and call it in every `it` case, to have a fresh store for each case.

Reply:
Ok, it's updated. I filed #846 to discuss testing stuff.